tests/unit/parsecmdline.js

```diff
@@ -0,0 +1,371 @@
+    assert.equal(args[0], -1e99);
+    assert.equal(Object.keys(params).length, 0);
+
+    args = []; params = {};
+    result = window.parseCommandLine("+1e99", args, params);
+    assert.equal(result, null);
+    assert.equal(args.length, 1);
+    assert.equal(args[0], 1e99);
+    assert.equal(Object.keys(params).length, 0);
+
+    args = []; params = {};
+    result = window.parseCommandLine("1e-99", args, params);
+    assert.equal(result, null);
+    assert.equal(args.length, 1);
+    assert.equal(args[0], 1e-99);
+    assert.equal(Object.keys(params).length, 0);
+
+    args = []; params = {};
+    result = window.parseCommandLine("1e+99", args, params);
+    assert.equal(result, null);
+    assert.equal(args.length, 1);
+    assert.equal(args[0], 1e99);
+    assert.equal(Object.keys(params).length, 0);
+
+    args = []; params = {};
+    result = window.parseCommandLine("1e999", args, params);
+    assert.equal(result, "Numeric argument has overflowed or is infinity");
+
+
+
+
+    // NULL
+
+    args = []; params = {};
+    result = window.parseCommandLine("null", args, params);
+    assert.isNull(result);
+    assert.equal(args.length, 1);
+    assert.equal(args[0], null);
+    assert.equal(Object.keys(params).length, 0);
+
+    args = []; params = {};
+    result = window.parseCommandLine("Null", args, params);
+    assert.isNull(result);
+    assert.equal(args.length, 1);
+    assert.equal(args[0], null);
+    assert.equal(Object.keys(params).length, 0);
+
+    args = []; params = {};
+    result = window.parseCommandLine("NULL", args, params);
+    assert.isNull(result);
+    assert.equal(args.length, 1);
+    assert.equal(args[0], null);
+    assert.equal(Object.keys(params).length, 0);
+
+    args = []; params = {};
+    result = window.parseCommandLine("NUll", args, params);
+    assert.isNull(result);
+    assert.equal(args.length, 1);
+    assert.equal(args[0], "NUll");
+    assert.equal(Object.keys(params).length, 0);
+
+
+
+    // NONE
+
+    args = []; params = {};
+    result = window.parseCommandLine("none", args, params);
+    assert.isNull(result);
+    assert.equal(args.length, 1);
+    assert.equal(args[0], "none");
+    assert.equal(Object.keys(params).length, 0);
+
+    args = []; params = {};
+    result = window.parseCommandLine("None", args, params);
+    assert.isNull(result);
+    assert.equal(args.length, 1);
+    assert.equal(args[0], null);
+    assert.equal(Object.keys(params).length, 0);
+
+    args = []; params = {};
+    result = window.parseCommandLine("NONE", args, params);
+
+
+    // GENERAL WHITESPACE HANDLING
+
+    assert.isNull(result);
+    assert.equal(args.length, 1);
+    assert.equal(args[0], "NONE");
+    assert.equal(Object.keys(params).length, 0);
+
+    args = []; params = {};
+    result = window.parseCommandLine("NOne", args, params);
+    assert.isNull(result);
+    assert.equal(args.length, 1);
+    assert.equal(args[0], "NOne");
+    assert.equal(Object.keys(params).length, 0);
+
```

Comment:
I know I've left similar comments above, but is there a reason for the newlines after comments, or the multiple lines between code blocks? I just think there is room for some style improvements 😄

Reply:
The unit tests in `tests/unit/parsecmdline.js` are grouped in sets. These comments are basically the group titles. 3 empty lines before that to visually separate from the previous group.

Reply:
I agree with oliver that there were 's a bit too much whitespaces. So I adjusted it a bit.